src/web/utils/firefoxUtils.ts

```diff
@@ -0,0 +1,26 @@
+import type {MarkdownTextInputElement} from '../../MarkdownTextInput.web';
+import {getCurrentCursorPosition, setCursorPosition} from './cursorUtils';
+
+function handleFirefoxRightArrowKeyNavigation(target: MarkdownTextInputElement, isSelectionEvent = false): void {
+  const currentSelection = getCurrentCursorPosition(target);
+  if (!currentSelection) {
+    return;
+  }
+
+  const text = target.value;
+  const cursorPos = currentSelection.end;
+
+  const segmenter = new Intl.Segmenter('en', {granularity: 'grapheme'});
+  const graphemes = Array.from(segmenter.segment(text));
```

Comment:
How long does this step take for longer values (approx 5000 characters)?

Can we somehow avoid allocating an array of graphemes and use the returned iterable instead?